src/Microsoft.AspNetCore.Mvc.Testing/MvcWebApplicationBuilder.cs

```diff
@@ -56,13 +61,16 @@ public MvcWebApplicationBuilder<TStartup> ConfigureAfterStartup(Action<IServiceC
         public MvcWebApplicationBuilder<TStartup> UseApplicationAssemblies()
         {
             var depsFileName = $"{typeof(TStartup).Assembly.GetName().Name}.deps.json";
-            var depsFile = new FileInfo(Path.Combine(Directory.GetCurrentDirectory(), depsFileName));
+            var depsFile = new FileInfo(Path.Combine(AppContext.BaseDirectory, depsFileName));
             if (!depsFile.Exists)
             {
-                throw new InvalidOperationException($"Can't find'{depsFile}'. This file is required for functional tests " +
-                    $"to run properly. There should be a copy of the file on your source project bin folder. If thats not the " +
-                    $"case, make sure that the property PreserveCompilationContext is set to true on your project file. E.g" +
-                    $"'<PreserveCompilationContext>true</PreserveCompilationContext>'.");
+                throw new InvalidOperationException($"Can't find'{depsFile.FullName}'. This file is required for functional tests " +
+                    "to run properly. There should be a copy of the file on your source project bin folder. If thats not the " +
+                    "case, make sure that the property PreserveCompilationContext is set to true on your project file. E.g" +
+                    "'<PreserveCompilationContext>true</PreserveCompilationContext>'." +
+                    $"For functional tests to work they need to either run from the build output folder or the deps file must be copied" +
+                    "to the folder where the tests are running on. A common case for this scenario is having shadow copying enabled when the" +
```

Comment:
A common ~~case~~ cause for this ~~scenario~~ error is running tests with shadow copy enabled.